packages/appsync-modelgen-plugin/src/__tests__/utils/process-has-many.test.ts

```diff
@@ -0,0 +1,382 @@
+import { CodeGenConnectionType, CodeGenFieldConnectionHasMany } from '../../utils/process-connections';
+import { hasManyHasImplicitKey, addHasManyKey } from '../../utils/process-has-many';
+import { CodeGenField, CodeGenModel } from '../../visitors/appsync-visitor';
+
+describe('hasManyHasImplicitKey', () => {
+  it('returns true for implicit primary key, no belongsTo', () => {
+    /**
+     * Example using the following schema:
+     *
+     * type Foo @model {
+     *   id: ID!
+     *   bar: [Bar] @hasMany
+     * }
+     *
+     * type Bar @model {
+     *   id: ID!
+     * }
+     */
+    const hasManyField: CodeGenField = {
+      name: 'bar',
+      type: 'Bar',
+      isList: true,
+      isNullable: true,
+      directives: [{ name: 'hasMany', arguments: {} }],
+    };
+    const fooModel: CodeGenModel = {
+      name: 'Foo',
+      type: 'model',
+      directives: [
+        { name: 'model', arguments: {} },
+      ],
+      fields: [
+        {
+          name: 'id',
+          type: 'ID',
+          isList: false,
+          isNullable: false,
+          directives: [],
+        },
+        hasManyField,
+      ],
+    };
+    const associationField: CodeGenField = {
+      name: 'fooBarsId',
+      type: 'ID',
+      isList: false,
+      isNullable: true,
+      directives: [],
+    };
+    const barModel: CodeGenModel = {
+      name: 'Bar',
+      type: 'model',
+      directives: [
+        { name: 'model', arguments: {} },
+      ],
+      fields: [
+        {
+          name: 'id',
+          type: 'ID',
+          isList: false,
+          isNullable: false,
+          directives: [],
+        },
+        associationField,
+      ],
+    }
+    const connection: CodeGenFieldConnectionHasMany = {
+      kind: CodeGenConnectionType.HAS_MANY,
+      connectedModel: barModel,
+      isConnectingFieldAutoCreated: true,
+      associatedWith: associationField,
+      associatedWithFields: [associationField],
+    };
+    expect(hasManyHasImplicitKey(hasManyField, fooModel, connection)).toBeTruthy();
```

Comment:
nit: here and in other places, it's better to avoid `toBeTruthy` since it means "type conversion of the value to boolean is true" and hence differs from `toBeTrue` or `toBe(true)`.

Reply:
I'll update.